### DISCUSSION TITLE
Should Azure DevOps autodiscover return disabled repositories?

### DISCUSSION THREAD

**Discussion by wterpstra** [renovatebot/renovate#25100] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.9.1

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm currently testing out the autodiscover feature of Renovate together with Azure DevOps. When outputting the autodiscovered repositories with the writeDiscoveredRepos feature I found that it also outputs repositories that are disabled. I was wondering if this is intended behavior. I don't see the added value for including these as you cannot check out these repositories. Can we filter out these values from the autodiscover result?

### What you tried so far.

Ran autodiscover, output the result. Then scale horizontally based on that output. When starting up an agent to renovate the disabled repository it will exit.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Repository is disabled- throwing error to abort renovation (repository=***/***)
 INFO: Repository is archived - skipping (repository=***/***)
DEBUG: Repository result: archived, status: disabled, enabled: false, onboarded: undefined (repository=***/***)

```

</details>

**Comment by rarkins** [comment]
If the result from autodiscover includes enough info to filter them out then we should. Otherwise the disabled repos will be found and quickly skipped during the run.

PR welcome if you think you can improve the current approach

**Comment by wterpstra** [comment]
Will do!

**Reply by wterpstra** [reply, marked as answer]
Done, see #25104
